services/api/src/resources/billing/resolvers.ts

```diff
@@ -0,0 +1,136 @@
+
+  // Action
+  return billingModel.addBillingModifier(input);
+};
+
+/**
+ * Update Billing Modifier
+ *
+ * @param {obj} root The rootValue passed from the Apollo server configuration.
+ * @param {UpdateBillingModifierInput} args
+ * @param {ExpressContext} context this includes the context passed from the apolloServer query
+ *     { sqlClient, hasPermissions, keycloakGrant, requestCache }
+ *
+ * @return {String} Success
+ */
+export const updateBillingModifier = async (_, args: UpdateBillingModifierInput, context: {models: any, hasPermission: any}) => {
+  const { hasPermission } = context;
+  const { input : { id, patch } } = args;
+
+  // Permissions
+  await hasPermission('group', 'update');
+
+  // Action
+  return billingModel.updateBillingModifier(id, patch);
+}
+
+/**
+ * Delete Billing Modifier
+ *
+ * @param {obj} root The rootValue passed from the Apollo server configuration.
+ * @param {DeleteBillingModifierInput} args
+ * @param {ExpressContext} context this includes the context passed from the apolloServer query
+ *     { sqlClient, hasPermissions, keycloakGrant, requestCache }
+ *
+ * @return {String} Success
+ */
+export const deleteBillingModifier = async (
+  _,
+  args: DeleteBillingModifierInput,
+  context: {models: any,
+  hasPermission: any}
+) => {
+  const { hasPermission } = context;
+  const { input: { id } } = args;
+
+  // Permissions
+  await hasPermission('group', 'delete');
+
+  // Action
+  return billingModel.deleteBillingModifier(id);
+}
+
+/**
+ * Delete All Billing Modifiers for a given Billing Gropu
+ *
+ * @param {obj} root The rootValue passed from the Apollo server configuration.
+ * @param {DeleteAllBillingGroupModifiersInput} args
+ * @param {ExpressContext} context this includes the context passed from the apolloServer query
+ *     { sqlClient, hasPermissions, keycloakGrant, requestCache }
+ *
+ * @return {String} Success
+ */
+export const deleteAllBillingModifiersByBillingGroup = async (
+  _,
+  args: DeleteAllBillingGroupModifiersInput,
+  context: {models: any,
+  hasPermission: any}
+) => {
+  const { hasPermission } = context;
+  const { input } = args;
+
+  // Permissions
+  await hasPermission('group', 'delete');
+
+
+  // Action
+  return billingModel.deleteAllBillingGroupModifiers(input);
+}
+
+/**
+ * Get All Billing Modifiers Added to a Billing Group
+ *
+ * @param {obj} root The rootValue passed from the Apollo server configuration.
+ * @param {BillingModifiersInput} args
+ * @param {ExpressContext} context this includes the context passed from the apolloServer query
+ *     { sqlClient, hasPermissions, keycloakGrant, requestCache }
+ *
+ * @return {[BillingModifier]} All modifiers associated to the billing Group
+ */
+export const getBillingModifiers = async (_, args: BillingModifiersInput, context: {models: any, hasPermission: any}) => {
+  const { hasPermission } = context;
+  const { input: groupInput, month } = args;
+
+  // Permissions
+  await hasPermission('group', 'view');
```

Comment:
There is no `view` scope for groups. All users who are a member of a group are assumed to be able to view it. So here, you'd want to check if the user can view groups they're not a member of (viewAll scope) and if not, check if they're a member of the group given in input. See [`getGroupsByProjectId`](https://github.com/amazeeio/lagoon/blob/master/services/api/src/resources/group/resolvers.ts#L63) for example

Reply:
Addressed in bcb60674748dc28ef2fa5f75027f9c618e78950c